movebucket_test.go

```diff
@@ -0,0 +1,482 @@
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTx_MoveBucket(t *testing.T) {
+	testCases := []struct {
+		name          string
+		srcBucketPath []string
+		dstBucketPath []string
+		bucketToMove  string
+		isSrcKey      bool
+		isDstKey      bool
+		expActErr     error
+	}{
+		{
+			name:          "happy path",
+			srcBucketPath: []string{"rootBucketX", "parentBucketY", "myBucketZ"},
+			dstBucketPath: []string{"rootBucketA", "parentBucketB"},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      false,
+			expActErr:     nil,
+		},
+		{
+			name:          "bucketToMove not exist in srcBucket",
+			srcBucketPath: []string{"rootBucketX", "parentBucketY"},
+			dstBucketPath: []string{"rootBucketA", "parentBucketB"},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      false,
+			expActErr:     errors.ErrBucketNotFound,
+		},
+		{
+			name:          "bucketToMove exist in dstBucket",
+			srcBucketPath: []string{"rootBucketX", "parentBucketY", "myBucketZ"},
+			dstBucketPath: []string{"rootBucketA", "parentBucketB", "myBucketZ"},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      false,
+			expActErr:     errors.ErrBucketExists,
+		},
+		{
+			name:          "bucketToMove key exist in srcBucket, but no subBucket value",
+			srcBucketPath: []string{"rootBucketX", "parentBucketY"},
+			dstBucketPath: []string{"rootBucketA", "parentBucketB"},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      true,
+			isDstKey:      false,
+			expActErr:     errors.ErrIncompatibleValue,
+		},
+		{
+			name:          "bucketToMove key exist in dstBucket, but no subBucket value",
+			srcBucketPath: []string{"rootBucketX", "parentBucketY", "myBucketZ"},
+			dstBucketPath: []string{"rootBucketA", "parentBucketB"},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      true,
+			expActErr:     errors.ErrIncompatibleValue,
+		},
+		{
+			name:          "srcBucket is rootBucket",
+			srcBucketPath: []string{"", "myBucketZ"},
+			dstBucketPath: []string{"rootBucketA", "parentBucketB"},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      false,
+			expActErr:     nil,
+		},
+		{
+			name:          "dstBucket is rootBucket",
+			srcBucketPath: []string{"rootBucketX", "parentBucketY", "myBucketZ"},
+			dstBucketPath: []string{""},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      false,
+			expActErr:     nil,
+		},
+		{
+			name:          "srcBucket is rootBucket, and dstBucket is rootBucket",
+			srcBucketPath: []string{"", "myBucketZ"},
+			dstBucketPath: []string{""},
+			bucketToMove:  "myBucketZ",
+			isSrcKey:      false,
+			isDstKey:      false,
+			expActErr:     errors.ErrSameBuckets,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		t.Run(tc.name, func(*testing.T) {
+			db := btesting.MustCreateDB(t)
+
+			dumpBucketBeforeMoving := tempfile()
+			dumpBucketAfterMoving := tempfile()
```

Comment:
Let's use `t.TempDir` instead of `tempfile`. Note the directory created by `t.TempDir` is automatically removed by Cleanup when the test and all its subtests complete.

Reply:
done